lib/sort_by_length.rb

```diff
@@ -1,7 +1,20 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
+#
+# interpretation of selection sort using .delete
+#
 def sort_by_length(my_sentence)
```

Comment:
Since you create a new array here this is O(n) space complexity.  Could you do a sort in-place?